lib/tasks/gitlab/check.rake

```diff
@@ -278,6 +279,29 @@ namespace :gitlab do
       end
     end
 
+    def check_upload_perms
+      print "Upload directory permissions correct? ... "
+
+      upload_path = Rails.root.join(Gitlab.config.gitlab.user_home,"gitlab-rails","uploads")
+      upload_path_tmp = Rails.root.join(upload_path,"tmp")
```

Comment:
For the second variable, I would write `File.join(upload_path, 'tmp')`. That path manipulation does not care about the Rails root.